src/Shared/ServerInfrastructure/HttpCharacters.cs

```diff
@@ -1,224 +1,91 @@
+    private const string AlphaNumeric = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+
+    // Authority https://tools.ietf.org/html/rfc3986#section-3.2
+    // Examples:
+    // microsoft.com
+    // hostname:8080
+    // [::]:8080
+    // [fe80::]
+    // 127.0.0.1
+    // user@host.com
+    // user:password@host.com
+    private static readonly IndexOfAnyValues<byte> _allowedAuthorityBytes = IndexOfAnyValues.Create(":.-[]@0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"u8);
+
+    // Matches Http.Sys
+    // Matches RFC 3986 except "*" / "+" / "," / ";" / "=" and "%" HEXDIG HEXDIG which are not allowed by Http.Sys
+    private static readonly IndexOfAnyValues<char> _allowedHostChars = IndexOfAnyValues.Create("!$&'()-._~" + AlphaNumeric);
+
+    // tchar https://tools.ietf.org/html/rfc7230#appendix-B
+    private static readonly IndexOfAnyValues<char> _allowedTokenChars = IndexOfAnyValues.Create("!#$%&'*+-.^_`|~" + AlphaNumeric);
+    private static readonly IndexOfAnyValues<byte> _allowedTokenBytes = IndexOfAnyValues.Create("!#$%&'*+-.^_`|~0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"u8);
+
+    // field-value https://tools.ietf.org/html/rfc7230#section-3.2
+    private static readonly IndexOfAnyValues<char> _allowedFieldChars = CreateAllowedFieldChars();
+
+    private static IndexOfAnyValues<char> CreateAllowedFieldChars()
     {
         // field-value https://tools.ietf.org/html/rfc7230#section-3.2
-        var fieldValue = new bool[_tableSize];
 
-        fieldValue[0x9] = true; // HTAB
+        Span<char> tmp = stackalloc char[128];
+        tmp[0] = (char)0x9;     // HTAB
+        var count = 1;
 
-        for (var c = 0x20; c <= 0x7e; c++) // VCHAR and SP
+        for (var c = 0x20; c <= 0x7E; ++c)  // VCHAR and SP
         {
-            fieldValue[c] = true;
+            tmp[count++] = (char)c;
         }
-        return fieldValue;
-    }
-
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static bool ContainsInvalidAuthorityChar(Span<byte> s)
-    {
-        var authority = _authority;
 
-        for (var i = 0; i < s.Length; i++)
-        {
-            var c = s[i];
-            if (c >= (uint)authority.Length || !authority[c])
-            {
-                return true;
-            }
-        }
-
-        return false;
+        Debug.Assert(count <= tmp.Length);
+        return IndexOfAnyValues.Create(tmp[..count]);
     }
 
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static int IndexOfInvalidHostChar(string s)
-    {
-        var host = _host;
-
-        for (var i = 0; i < s.Length; i++)
-        {
-            var c = s[i];
-            if (c >= (uint)host.Length || !host[c])
-            {
-                return i;
-            }
-        }
+    public static bool ContainsInvalidAuthorityChar(ReadOnlySpan<byte> span) => span.IndexOfAnyExcept(_allowedAuthorityBytes) >= 0;
```

Comment:
All of these helpers had `AggressiveInlining` before. I guess it wouldn't hurt to keep them - it's even less code for the caller than before.

Reply:
I removed them as now it's one-liners and the JIT will inline these calls anyway.
Do you still think it's worth it to bring the attributes back?

Reply:
If you think they'll get inlined anyway then I guess we don't need them